There are many warning logs in paddle CI.

```
04:39:10]W:	 [Step 1/1] /paddle/build/third_party/anakin/src/extern_anakin/utils/logger/logger_core.h:761:85: warning: field width specifier '*' expects argument of type 'int', but argument 4 has type 'long long unsigned int' [-Wformat=]
[04:39:10]W:	 [Step 1/1] In file included from /paddle/build/third_party/anakin/src/extern_anakin/utils/logger/logger.h:22:0,
[04:39:10]W:	 [Step 1/1]                  from /paddle/build/third_party/anakin/src/extern_anakin/framework/core/parameter.h:24,
[04:39:10]W:	 [Step 1/1]                  from /paddle/build/third_party/anakin/src/extern_anakin/framework/core/data_types.h:19,
[04:39:10]W:	 [Step 1/1]                  from /paddle/build/third_party/anakin/src/extern_anakin/framework/operators/flatten.h:20,
[04:39:10]W:	 [Step 1/1]                  from /paddle/build/third_party/anakin/src/extern_anakin/framework/operators/flatten.cpp:1:
[04:39:10]W:	 [Step 1/1] /paddle/build/third_party/anakin/src/extern_anakin/utils/logger/logger_core.h: In function 'void logger::core::funcRegister::get_thread_name(char*, long long unsigned int, bool)':
[04:39:10]W:	 [Step 1/1] /paddle/build/third_party/anakin/src/extern_anakin/utils/logger/logger_core.h:761:85: warning: field width specifier '*' expects argument of type 'int', but argument 4 has type 'long long unsigned int' [-Wformat=]
[04:39:10]W:	 [Step 1/1]          snprintf(buffer, length, "%*X", length - 1, static_cast<unsigned>(thread_id));
[04:39:10]W:	 [Step 1/1]                                                                                      ^
[04:39:10]W:	 [Step 1/1] /paddle/build/third_party/anakin/src/extern_anakin/utils/logger/logger_core.h:761:85: warning: field width specifier '*' expects argument of type 'int', but argument 4 has type 'long long unsigned int' [-Wformat=]
[04:39:10] :	 [Step 1/1] [ 78%] Building CXX object framework/CMakeFiles/anakin.dir/operators/gru.cpp.o
[04:39:10] :	 [Step 1/1] [ 19%] Linking CXX static library libcontext_project.a
[04:39:10] :	 [Step 1/1] [ 19%] Built target context_project
[04:39:10] :	 [Step 1/1] [ 79%] Building CXX object framework/CMakeFiles/anakin.dir/operators/im2sequence.cpp.o
[04:39:10]W:	 [Step 1/1] In file included from /paddle/build/third_party/anakin/src/extern_anakin/utils/logger/logger.h:22:0,
[04:39:10]W:	 [Step 1/1]                  from /paddle/build/third_party/anakin/src/extern_anakin/framework/core/parameter.h:24,
[04:39:10]W:	 [Step 1/1]                  from /paddle/build/third_party/anakin/src/extern_anakin/framework/core/data_types.h:19,
[04:39:10]W:	 [Step 1/1]                  from /paddle/build/third_party/anakin/src/extern_anakin/framework/operators/gather.h:20,
[04:39:10]W:	 [Step 1/1]                  from /paddle/build/third_party/anakin/src/extern_anakin/framework/operators/gather.cpp:1:
[04:39:10]W:	 [Step 1/1] /paddle/build/third_party/anakin/src/extern_anakin/utils/logger/logger_core.h: In function 'void logger::core::funcRegister::get_thread_name(char*, long long unsigned int, bool)':
[04:39:10]W:	 [Step 1/1] /paddle/build/third_party/anakin/src/extern_anakin/utils/logger/logger_core.h:761:85: warning: field width specifier '*' expects argument of type 'int', but argument 4 has type 'long long unsigned int' [-Wformat=]
[04:39:10]W:	 [Step 1/1]          snprintf(buffer, length, "%*X", length - 1, static_cast<unsigned>(thread_id));
[04:39:10]W:	 [Step 1/1]                                                                                      ^
```


```
[04:38:09]W:	 [Step 1/1] /paddle/build/third_party/install/protobuf/include/google/protobuf/map.h: In instantiation of 'google::protobuf::Map<Key, T>::size_type google::protobuf::Map<Key, T>::InnerMap::TableSize(google::protobuf::Map<Key, T>::size_type) [with Key = std::__cxx11::basic_string<char>; T = List; google::protobuf::Map<Key, T>::size_type = long unsigned int]':
[04:38:09]W:	 [Step 1/1] /paddle/build/third_party/install/protobuf/include/google/protobuf/map.h:730:20:   required from 'google::protobuf::Map<Key, T>::InnerMap::InnerMap(google::protobuf::Map<Key, T>::size_type, google::protobuf::Map<Key, T>::hasher, google::protobuf::Map<Key, T>::Allocator) [with Key = std::__cxx11::basic_string<char>; T = List; google::protobuf::Map<Key, T>::size_type = long unsigned int; google::protobuf::Map<Key, T>::hasher = google::protobuf::hash<std::__cxx11::basic_string<char> >; google::protobuf::Map<Key, T>::Allocator = google::protobuf::Map<std::__cxx11::basic_string<char>, List>::MapAllocator<google::protobuf::Map<std::__cxx11::basic_string<char>, List>::KeyValuePair>]'
[04:38:09]W:	 [Step 1/1] /paddle/build/third_party/install/protobuf/include/google/protobuf/arena.h:344:36:   required from 'static T* google::protobuf::Arena::Create(google::protobuf::Arena*, const Arg1&, const Arg2&, const Arg3&) [with T = google::protobuf::Map<std::__cxx11::basic_string<char>, List>::InnerMap; Arg1 = int; Arg2 = google::protobuf::hash<std::__cxx11::basic_string<char> >; Arg3 = google::protobuf::Map<std::__cxx11::basic_string<char>, List>::MapAllocator<google::protobuf::Map<std::__cxx11::basic_string<char>, List>::KeyValuePair>]'
[04:38:09]W:	 [Step 1/1] /paddle/build/third_party/install/protobuf/include/google/protobuf/map.h:572:34:   required from 'void google::protobuf::Map<Key, T>::Init() [with Key = std::__cxx11::basic_string<char>; T = List]'
[04:38:09]W:	 [Step 1/1] /paddle/build/third_party/install/protobuf/include/google/protobuf/map.h:530:9:   required from 'google::protobuf::Map<Key, T>::Map(bool) [with Key = std::__cxx11::basic_string<char>; T = List]'
[04:38:09]W:	 [Step 1/1] /paddle/build/third_party/install/protobuf/include/google/protobuf/map_field_lite.h:101:8:   required from 'google::protobuf::internal::MapFieldLite<Key, T, key_wire_type, value_wire_type, default_enum_value>::MapFieldLite() [with Key = std::__cxx11::basic_string<char>; T = List; google::protobuf::internal::WireFormatLite::FieldType key_wire_type = (google::protobuf::internal::WireFormatLite::FieldType)9u; google::protobuf::internal::WireFormatLite::FieldType value_wire_type = (google::protobuf::internal::WireFormatLite::FieldType)11u; int default_enum_value = 0]'
[04:38:09]W:	 [Step 1/1] /paddle/build/third_party/install/protobuf/include/google/protobuf/map_field_inl.h:170:26:   required from 'google::protobuf::internal::MapField<Key, T, key_wire_type, value_wire_type, default_enum_value>::MapField() [with Key = std::__cxx11::basic_string<char>; T = List; google::protobuf::internal::WireFormatLite::FieldType kKeyFieldType = (google::protobuf::internal::WireFormatLite::FieldType)9u; google::protobuf::internal::WireFormatLite::FieldType kValueFieldType = (google::protobuf::internal::WireFormatLite::FieldType)11u; int default_enum_value = 0]'
[04:38:09]W:	 [Step 1/1] /paddle/build/third_party/anakin/src/extern_anakin/framework/model_parser/proto/graph.pb.cpp:1458:60:   required from here
[04:38:09]W:	 [Step 1/1] /paddle/build/third_party/install/protobuf/include/google/protobuf/map.h:1292:32: warning: enumeral and non-enumeral type in conditional expression [-Wextra]
[04:38:09]W:	 [Step 1/1]        return n < kMinTableSize ? kMinTableSize : n;
[04:38:09]W:	 [Step 1/1]                                 ^
[04:38:09]W:	 [Step 1/1] /paddle/build/third_party/install/protobuf/include/google/protobuf/map.h: In instantiation of 'google::protobuf::Map<Key, T>::size_type google::protobuf::Map<Key, T>::InnerMap::TableSize(google::protobuf::Map<Key, T>::size_type) [with Key = std::__cxx11::basic_string<char>; T = TensorProto; google::protobuf::Map<Key, T>::size_type = long unsigned int]':
[04:38:09]W:	 [Step 1/1] /paddle/build/third_party/install/protobuf/include/google/protobuf/map.h:730:20:   required from 'google::protobuf::Map<Key, T>::InnerMap::InnerMap(google::protobuf::Map<Key, T>::size_type, google::protobuf::Map<Key, T>::hasher, google::protobuf::Map<Key, T>::Allocator) [with Key = std::__cxx11::basic_string<char>; T = TensorProto; google::protobuf::Map<Key, T>::size_type = long unsigned int; google::protobuf::Map<Key, T>::hasher = google::protobuf::hash<std::__cxx11::basic_string<char> >; google::protobuf::Map<Key, T>::Allocator = google::protobuf::Map<std::__cxx11::basic_string<char>, TensorProto>::MapAllocator<google::protobuf::Map<std::__cxx11::basic_string<char>, TensorProto>::KeyValuePair>]'
[04:38:09]W:	 [Step 1/1] /paddle/build/third_party/install/protobuf/include/google/protobuf/arena.h:344:36:   required from 'static T* google::protobuf::Arena::Create(google::protobuf::Arena*, const Arg1&, const Arg2&, const Arg3&) [with T = google::protobuf::Map<std::__cxx11::basic_string<char>, TensorProto>::InnerMap; Arg1 = int; Arg2 = google::protobuf::hash<std::__cxx11::basic_string<char> >; Arg3 = google::protobuf::Map<std::__cxx11::basic_string<char>, TensorProto>::MapAllocator<google::protobuf::Map<std::__cxx11::basic_string<char>, TensorProto>::KeyValuePair>]'
[04:38:09]W:	 [Step 1/1] /paddle/build/third_party/install/protobuf/include/google/protobuf/map.h:572:34:   required from 'void google::protobuf::Map<Key, T>::Init() [with Key = std::__cxx11::basic_string<char>; T = TensorProto]'
[04:38:09]W:	 [Step 1/1] /paddle/build/third_party/install/protobuf/include/google/protobuf/map.h:530:9:   required from 'google::protobuf::Map<Key, T>::Map(bool) [with Key = std::__cxx11::basic_string<char>; T = TensorProto]'
[04:38:09]W:	 [Step 1/1] /paddle/build/third_party/install/protobuf/include/google/protobuf/map_field_lite.h:101:8:   required from 'google::protobuf::internal::MapFieldLite<Key, T, key_wire_type, value_wire_type, default_enum_value>::MapFieldLite() [with Key = std::__cxx11::basic_string<char>; T = TensorProto; google::protobuf::internal::WireFormatLite::FieldType key_wire_type = (google::protobuf::internal::WireFormatLite::FieldType)9u; google::protobuf::internal::WireFormatLite::FieldType value_wire_type = (google::protobuf::internal::WireFormatLite::FieldType)11u; int default_enum_value = 0]'
[04:38:09]W:	 [Step 1/1] /paddle/build/third_party/install/protobuf/include/google/protobuf/map_field_inl.h:170:26:   required from 'google::protobuf::internal::MapField<Key, T, key_wire_type, value_wire_type, default_enum_value>::MapField() [with Key = std::__cxx11::basic_string<char>; T = TensorProto; google::protobuf::internal::WireFormatLite::FieldType kKeyFieldType = (google::protobuf::internal::WireFormatLite::FieldType)9u; google::protobuf::internal::WireFormatLite::FieldType kValueFieldType = (google::protobuf::internal::WireFormatLite::FieldType)11u; int default_enum_value = 0]'
```

--- a/cmake/external/anakin.cmake
+++ b/cmake/external/anakin.cmake
@@ -48,7 +48,7 @@ ExternalProject_Add(
     ${EXTERNAL_PROJECT_LOG_ARGS}
     DEPENDS             ${MKLML_PROJECT}
     GIT_REPOSITORY      "https://github.com/PaddlePaddle/Anakin"
-    GIT_TAG             "9424277cf9ae180a14aff09560d3cd60a49c76d2"
+    GIT_TAG             "3c8554f4978628183566ab7dd6c1e7e66493c7cd"
     PREFIX              ${ANAKIN_SOURCE_DIR}
     UPDATE_COMMAND      ""
     CMAKE_ARGS          ${CMAKE_ARGS_PREFIX}

--- a/cmake/inference_lib.cmake
+++ b/cmake/inference_lib.cmake
@@ -150,7 +150,7 @@ if (WITH_ANAKIN AND WITH_MKL)
         SRCS
         ${PADDLE_BINARY_DIR}/paddle/fluid/inference/api/libinference_anakin_api* # compiled anakin api
         ${ANAKIN_INSTALL_DIR} # anakin release
-        DSTS ${dst_dir}/inference/anakin ${dst_dir}/inference/anakin)
+        DSTS ${dst_dir}/inference/anakin ${FLUID_INSTALL_DIR}/third_party/install/anakin)
      list(APPEND inference_deps anakin_inference_lib)
 endif()
 

--- a/paddle/fluid/inference/api/api_anakin_engine_rnn_tester.cc
+++ b/paddle/fluid/inference/api/api_anakin_engine_rnn_tester.cc
@@ -20,70 +20,15 @@ limitations under the License. */
 #include <iostream>
 #include <thread>  // NOLINT
 #include <vector>
-#include "framework/core/net/net.h"
+#include "paddle/fluid/inference/api/helper.h"
 #include "paddle/fluid/inference/api/paddle_inference_api.h"
+#include "paddle/fluid/inference/api/timer.h"
+#include "utils/logger/logger.h"
 
 DEFINE_string(model, "", "Directory of the inference model.");
 DEFINE_string(datapath, "", "Path of the dataset.");
 DEFINE_int32(batch_size, 1, "batch size.");
 DEFINE_int32(repeat, 1, "Running the inference program repeat times.");
-
-// Timer for timer
-class Timer {
- public:
-  double start;
-  double startu;
-  void tic() {
-    struct timeval tp;
-    gettimeofday(&tp, NULL);
-    start = tp.tv_sec;
-    startu = tp.tv_usec;
-  }
-  double toc() {
-    struct timeval tp;
-    gettimeofday(&tp, NULL);
-    double used_time_ms =
-        (tp.tv_sec - start) * 1000.0 + (tp.tv_usec - startu) / 1000.0;
-    return used_time_ms;
-  }
-};
-
-std::vector<std::string> string_split(std::string in_str,
-                                      std::string delimiter) {
-  std::vector<std::string> seq;
-  int found = in_str.find(delimiter);
-  int pre_found = -1;
-  while (found != std::string::npos) {
-    if (pre_found == -1) {
-      seq.push_back(in_str.substr(0, found));
-    } else {
-      seq.push_back(in_str.substr(pre_found + delimiter.length(),
-                                  found - delimiter.length() - pre_found));
-    }
-    pre_found = found;
-    found = in_str.find(delimiter, pre_found + delimiter.length());
-  }
-  seq.push_back(
-      in_str.substr(pre_found + 1, in_str.length() - (pre_found + 1)));
-  return seq;
-}
-std::vector<std::string> string_split(
-    std::string in_str, std::vector<std::string>& delimiter) {  // NOLINT
-  std::vector<std::string> in;
-  std::vector<std::string> out;
-  out.push_back(in_str);
-  for (auto del : delimiter) {
-    in = out;
-    out.clear();
-    for (auto s : in) {
-      auto out_s = string_split(s, del);
-      for (auto o : out_s) {
-        out.push_back(o);
-      }
-    }
-  }
-  return out;
-}
 
 class Data {
  public:
@@ -120,36 +65,24 @@ void Data::get_batch_data(
   week_fea.clear();
   time_fea.clear();
   while (_file.getline(buf, 10000)) {
-    std::string s = buf;
-    std::vector<std::string> deli_vec = {":"};
-    std::vector<std::string> data_vec = string_split(s, deli_vec);
+    std::vector<std::string> data_vec;
+    paddle::inference::split(buf, ':', &data_vec);
 
     std::vector<std::string> seq;
-    seq = string_split(data_vec[0], {"|"});
+    paddle::inference::split(data_vec[0], '|', &seq);
 
     for (auto link : seq) {
-      std::vector<std::string> data = string_split(link, ",");
       std::vector<float> vec;
-      for (int i = 0; i < data.size(); i++) {
-        vec.push_back(atof(data[i].c_str()));
-      }
+      paddle::inference::split_to_float(link, ',', &vec);
       fea.push_back(vec);
     }
-    std::vector<std::string> week_data;
-    std::vector<std::string> time_data;
 
-    week_data = string_split(data_vec[2], ",");
     std::vector<float> vec_w;
-    for (int i = 0; i < week_data.size(); i++) {
-      vec_w.push_back(atof(week_data[i].c_str()));
-    }
+    paddle::inference::split_to_float(data_vec[2], ',', &vec_w);
     week_fea.push_back(vec_w);
 
-    time_data = string_split(data_vec[1], ",");
     std::vector<float> vec_t;
-    for (int i = 0; i < time_data.size(); i++) {
-      vec_t.push_back(atof(time_data[i].c_str()));
-    }
+    paddle::inference::split_to_float(data_vec[1], ',', &vec_t);
     time_fea.push_back(vec_t);
 
     cum += seq.size();
@@ -275,14 +208,13 @@ void single_test() {
     inputs.push_back(tensor_2);
     inputs.push_back(tensor_0);
 
-    Timer timer;
+    paddle::inference::Timer timer;
     timer.tic();
     for (int i = 0; i < FLAGS_repeat; i++) predictor->Run(inputs, &outputs);
 
-    LOG(INFO) << "batch_size = " << FLAGS_batch_size
-              << ", repeat = " << FLAGS_repeat
-              << ", sequence_length = " << seq_offset[seq_offset.size() - 1]
-              << ", latency: " << timer.toc() / FLAGS_repeat << "ms";
+    paddle::inference::PrintTime(FLAGS_batch_size, FLAGS_repeat, 1, 0,
+                                 timer.toc() / FLAGS_repeat);
+    LOG(INFO) << "sequence_length = " << seq_offset[seq_offset.size() - 1];
 
     float* data_o = static_cast<float*>(outputs[0].data.data());
     VLOG(3) << "outputs[0].data.length() = " << outputs[0].data.length();

--- a/paddle/scripts/paddle_build.sh
+++ b/paddle/scripts/paddle_build.sh
@@ -547,7 +547,7 @@ function gen_capi_package() {
         rm -rf $install_prefix
         make DESTDIR="$install_prefix" install
         cd $install_prefix/usr/local
-        ls | egrep -v "^Found.*item$" | xargs tar -cf ${PADDLE_ROOT}/build/paddle.tgz
+        ls | egrep -v "^Found.*item$" | xargs tar -czf ${PADDLE_ROOT}/build/paddle.tgz
     fi
 }
 

--- a/python/paddle/fluid/layers/nn.py
+++ b/python/paddle/fluid/layers/nn.py
@@ -4500,7 +4500,7 @@ def reshape(x, shape, actual_shape=None, act=None, inplace=True, name=None):
     """
 
     if not (isinstance(shape, list) or isinstance(shape, tuple)):
-        raise ValueError("Input shape must be a python lsit or tuple.")
+        raise ValueError("Input shape must be a python list or tuple.")
     inputs = {"X": x}
     if isinstance(actual_shape, Variable):
         inputs["Shape"] = actual_shape

